server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -1054,7 +1061,7 @@ public void validateCustomParameters(ServiceOfferingVO serviceOffering, Map<Stri
 
             if (serviceOffering.getSpeed() == null) {
                 String cpuSpeed = customParameters.get(UsageEventVO.DynamicParameters.cpuSpeed.name());
-                if ((cpuSpeed == null) || (NumbersUtil.parseInt(cpuSpeed, -1) <= 0)) {
+                if ((cpuSpeed == null) || (parseInt(cpuSpeed, -1) <= 0)) {
```

Comment:
Could you please update the `cpuSpeed == null` check to `StringUtils.isNotBlank(cpuSpeed)`?

The `lang3.StringUtils.isNotBlank(CharSequence cs)` checks if a CharSequence is not empty (""), not null and not whitespace only.
```
 StringUtils.isNotBlank(null)      = false
 StringUtils.isNotBlank("")        = false
 StringUtils.isNotBlank(" ")       = false
 StringUtils.isNotBlank("bob")     = true
 StringUtils.isNotBlank("  bob  ") = true
```

Reply:
I did not change this code (only static import), but I adapt that to StringUtils.isBlank(). Thanks for the hint.